azure-keyvault/azure/keyvault/keys/_models.py

```diff
@@ -1,406 +1,216 @@
-    :ivar managed: True if the key's lifetime is managed by key vault. If this
-     is a key backing a certificate, then managed will be true.
-    :vartype managed: bool
-    """
+    @property
+    def updated(self):
+        # type: () -> datetime
+        """The Key's last updated time in UTC
+        :rtype: datetime"""
+        return self._attributes.updated
 
-    _validation = {"managed": {"readonly": True}}
+    @property
+    def vault_url(self):
+        # type: () -> str
+        """The url of the vault containing the key
+        :rtype: str"""
+        return self._vault_id.vault_url
 
-    _attribute_map = {
-        "kid": {"key": "kid", "type": "str"},
-        "attributes": {"key": "attributes", "type": "KeyAttributes"},
-        "tags": {"key": "tags", "type": "{str}"},
-        "managed": {"key": "managed", "type": "bool"},
-    }
+    @property
+    def recovery_level(self):
+        # type: () -> str
+        """The vault's deletion recovery level for keys
+        :rtype: str"""
+        return self._attributes.recovery_level
 
-    def __init__(self, **kwargs):
-        super(KeyItem, self).__init__(**kwargs)
-        self.kid = kwargs.get("kid", None)
-        self.attributes = kwargs.get("attributes", None)
-        self.tags = kwargs.get("tags", None)
-        self.managed = None
+    @property
+    def tags(self):
+        # type: () -> Dict[str, str]
+        """Application specific metadata in the form of key-value pairs.
+        :rtype: dict"""
+        return self._tags
 
 
-class DeletedKeyItem(KeyItem):
-    """The deleted key item containing the deleted key metadata and information
-    about deletion.
-    Variables are only populated by the server, and will be ignored when
-    sending a request.
-    :param kid: Key identifier.
-    :type kid: str
-    :param attributes: The key management attributes.
-    :type attributes: ~azure.keyvault.v7_0.models.KeyAttributes
-    :param tags: Application specific metadata in the form of key-value pairs.
-    :type tags: dict[str, str]
-    :ivar managed: True if the key's lifetime is managed by key vault. If this
-     is a key backing a certificate, then managed will be true.
-    :vartype managed: bool
-    :param recovery_id: The url of the recovery object, used to identify and
-     recover the deleted key.
-    :type recovery_id: str
-    :ivar scheduled_purge_date: The time when the key is scheduled to be
-     purged, in UTC
-    :vartype scheduled_purge_date: datetime
-    :ivar deleted_date: The time when the key was deleted, in UTC
-    :vartype deleted_date: datetime
+class Key(KeyBase):
+    """A key consisting of all KeyBase and key_material information.
     """
 
-    _validation = {
-        "managed": {"readonly": True},
-        "scheduled_purge_date": {"readonly": True},
-        "deleted_date": {"readonly": True},
-    }
+    def __init__(self, attributes, key, **kwargs):
+        super(Key, self).__init__(attributes, key.kid, **kwargs)
```

Comment:
Missing types, and the `key` parameter confuses me because we're constructing a `Key` here.

Reply:
`key` parameter in the init is used to assign it to `key_material` in the next line [here](https://github.com/samvaity/azure-sdk-for-python/blob/key-client-docs-test/azure-keyvault/azure/keyvault/keys/_models.py#L131)